### ISSUE TITLE
Doesn't CD into the directory afterward

### ISSUE THREAD

**Issue by mattstauffer** [tightenco/lambo#1]
¯(°_o)/¯ no idea now

**Comment by DanielDarrenJones** [comment]
Super hacky solution but you could use: 
```
open -a Terminal $PROJECTNAME
```

This will open a new terminal window and cd to the editor, although this means you are stuck using terminal and it opens a new window instead of the current terminal window, which isn't very elegant...

Annoyingly the bash commands are run in a sub-process which doesn't effect the main process, this is why the cd is not being persisted when the script is done.

Another possible improvement would be to include:
```
echo "cd $PROJECTNAME" | pbcopy
```
which adds the cd command to the clipboard so the user can just paste when the command is done.

**Comment by mattstauffer** [comment]
@DanielDarrenJones Good thinking on opening the terminal. Might be worth trying something like that.. hacky but worth a consideration.

And yah, I had read up on the sub-process.

The pbcopy is a good idea--but I carefully guard my clipboard (I know, I know), and I think I'd get pretty annoyed if it got overwritten randomly by a CLI script.

For now I'll keep it just with a notice at the end... will keep thinking and asking for input here. But if all else fails something along the lines of the `open` script might be worth consideration.

Thanks!!

**Comment by marklj** [comment]
*If* somehow `cd` made it in, it would open up the possibility to run `npm install` at the very end. I find myself doing this:

`lambo my-app; cd my-app; npm install`

This way, I'm ready to run Elixir *almost* immediately.

**Comment by milose** [comment]
Add this to your bash profile or wherever nowadays you store your terminal functions / aliases.

```bash
function lmb() {
    lambo $1 "${@:2}"
    cd $1
    yarn
}
alias lmb=lmb
```
Usage:
```bash
lmb [arguments]
```
It creates the project, `cd`'s into it and runs `yarn`.

**Comment by quickliketurtle** [comment]
Another option: 
Since sub-processes can affect child processes by opening another shell it will keep the current directory of the lambo process. 

We could add ```exec bash``` to open another bash process, but I think it's better to use the users login shell which is stored in the $SHELL variable. 
So if we add
```exec $SHELL```
around line 175 it has the desired effect.

I'd expect $SHELL to have a value for everyone, but we could check that it is valid then echo the current statement if it is not.

Let me know your thoughts.

**Comment by milose** [comment]
On my Ubuntu box everything works fine. However, on my Mac:

![image](https://cloud.githubusercontent.com/assets/2998029/23088773/0fb1c124-f580-11e6-8df5-7361d2c3a251.png)

**Comment by quickliketurtle** [comment]
@milose Looks like your shell is set to bash, but in your .bash_profile you are changing to zsh. Can you confirm? 
You would need your login shell to be set to the same as the shell you use for this to work. 
```chsh -s /bin/zsh``` will do that. 

Let me know if your environment is different than I'm assuming above. Thanks

**Comment by milose** [comment]
No, I use [`bash-it`](https://github.com/Bash-it/bash-it) Here is a gist of my `.bash_profile`

```bash
#!/usr/bin/env bash

export BASH_IT="/Users/milos/.bash_it"
export BASH_IT_THEME='bakke-mac'
source $BASH_IT/bash_it.sh

# VAR PATH
PATH="/usr/local/sbin:/usr/local/bin:$PATH"
PATH="/usr/local/bin/platformio:$PATH"
PATH="/Users/milos/.composer/vendor/laravel/valet:$PATH"
PATH="/Users/milos/.composer/vendor/bin:$PATH"

# Set my editor and git editor
export EDITOR="atom"
export GIT_EDITOR='atom'
```

**Comment by quickliketurtle** [comment]
Hi @milose, thanks for providing that.  What I'm reading points to a difference in how config files are loaded based on a bash login shell vs an interactive non-login shell. 

> login bash shells source either .bash_profile, .bash_login, or .profile; non-login interactive bash shells (e.g. subshells) source .bashrc instead.

So there are 2 solutions for you: 
1. Source your .bash_profile from .bashrc.
2. Move your config to .bashrc and source .bashrc from .bash_profile.
``` if [ -f ~/.bashrc ]; then . ~/.bashrc; fi ```

Option 2 seems to be the 'correct' way, if there ever is such a thing :-);

[From this post](http://stackoverflow.com/questions/4493063/best-practice-for-bash-start-up-files-on-a-mac/4493499#4493499)

Thanks.
--Jeff

**Comment by milose** [comment]
Thanks Jeff. I created `.bashrc` with:
```bash
if [ -f ~/.bash_profile ]; then source ~/.bash_profile; fi
```
After that it seems to work fine. I chose this option since some installer might opt to append to `.bash_profile` after detecting it is being run on a Mac and I want to have my configuration contained in just one file.

**Comment by mattstauffer** [comment]
Closed by #27
